Fixes foods triggering allergies when they really shouldn't and makes stuff made using the food processor inherit reagents

VladinXXV opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Removes as many reduntant medicines from food as I could find. These are meds in foods that either have no business being there, or would typically be inherited from one of the ingredients. This allows botany and the chef to work together to make foods safe to consume for everyone. This does NOT touch things like brain burgers, brain cake, arnold pizza, donk pockets, lollipops, and gumballs.

This also makes it so putting things into the food processor preserves reagents. Because carrot fries. They're why we're here right now. Thank carrot fries. I have no idea how I got it to work but I did.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
I'll give you an example that actually happened once. Split pea soup. It is made using carrots. Carrots naturally contain oculine. I had someone eat my soup and die to their oculine allergy. Okay. I removed oculine from my carrots, problem solved, right? No. Split pea soup contains oculine regardless of if you remove it from your carrots. So they died again. It's unintuitive and stupid. If I removed the oculine from the carrots, why does the soup have oculine? Where is it coming from? Why must those with allergies suffer, no matter what is done to try and fix it?
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl: Vladoricious
fix: The Paranormal Response team has successfully defeated the demon that was putting medicines in food from nowhere.
fix: The food processor no longer strips chemicals from the things you put inside. Finally, meth in my carrot fries.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

VladinXXV committed:
The Great De-Medicine-ing

fuck you and your reduntant chems

VladinXXV committed:
missed one

nice try

VladinXXV committed:
processor now inherits chems

holy shit guys i did it it works how did i do this i dont understand code at all oh my god this was so painful to do ahahahahahahahahahahahaha FUCK CARROT FRIES THIS IS ALL THEIR FAULT.

Palpetinus commented:
I'm sure this will upset **someone** who used uh..
_flips through the commit_
Salad of Eden for earthblood?

VladinXXV commented:
> I'm sure this will upset **someone** who used uh.. _flips through the commit_ Salad of Eden for earthblood?

The ingredients for Salad of Eden normally include earthsblood. Unless botany has removed the gene from the ambrosia gaia, it'll be passed along.

ATH1909 commented:
What happens if these food items are spawned in directly from, say, a silver extract? Will these carrot-based foods not contain oculine unless you craft them? Will the ambrosia-based ones not contain earthsblood unless you craft them?

Or is that not how it works?

VladinXXV commented:
> What happens if these food items are spawned in directly from, say, a silver extract? Will these carrot-based foods not contain oculine unless you craft them? Will the ambrosia-based ones not contain earthsblood unless you craft them?
> 
> Or is that not how it works?

Correct, they won't contain them unless crafted.

RaveRadbury commented on code/game/objects/items/food/soup.dm:
```diff
@@ -81,7 +81,7 @@
 	name = "nettle soup"
 	desc = "To think, the botanist would've beat you to death with one of these."
 	icon_state = "nettlesoup"
-	food_reagents = list(/datum/reagent/consumable/nutriment = 5, /datum/reagent/consumable/nutriment/protein = 2, /datum/reagent/water = 5, /datum/reagent/consumable/nutriment/vitamin = 9, /datum/reagent/medicine/omnizine = 5)
```
Nettle Soup doesn't inherit omnizine from any of its components, it seems to be a joke/reference to nettle soup being healthy irl. 

You can still grind up nettle soup and get the omnizine removed. It's a bit inelegant as a solution but it is a kind of small annoyance that you can deal with via future tech (chem master), which is the kind of wrench in the system that quirks are meant to create.

The other cases here seem to be redundant inheritance, but this is one case where the ingredients come together to make something that's more than the sum of their parts, which is neat.

VladinXXV replied:
If I'm gonna be completely honest: I didn't even know nettles were a real plant, let alone soup ingredient.

If you think it's better to leave in the omnizine, I'll make that change, but I don't know if "healthy irl" is exactly "5u of literal medicine". Granted, "carrots good for the eyes" translated into oculine, so.... Anyhow. I think the high vitamin content already covers being "healthy."

RaveRadbury replied:
Your PR body details that the point of doing this was so that people could intentionally grow plants without a chemical to make sure it doesn't go in. Imo it should do only that, and save the potential issue of emergent chemicals (of which we only have this case rn) for another day.

VladinXXV committed:
nettle soup has been spared

you mean to tell me nettles are a real plant and people EAT them???

MrMelbert commented on code/modules/food_and_drinks/kitchen_machinery/processor.dm:
```diff
@@ -58,6 +58,8 @@
 		var/cached_multiplier = (recipe.food_multiplier * rating_amount)
 		for(var/i in 1 to cached_multiplier)
 			var/atom/processed_food = new recipe.output(drop_location())
+			var/atom/input_food = new recipe.input
+			input_food.reagents.copy_to(processed_food, input_food.reagents.total_volume)
```
What happens to `input_food` after the proc is done?
Does it stay in nullspace? You may need to clean it up afterwards

VladinXXV replied:
`input_food` has been replaced with `what`

`what` is already cleaned at the end of this proc

all is well

VladinXXV committed:
what

shitty name for a variable if you ask me but at least i'm not making a variable i never needed to now

VladinXXV committed:
no your honor, i was not duping carrot oculine, that'd be absurd

you can't prove anything. i plead the fifth.

BallastMonster commented:
huh I was wondering how I instantly fucking died, thanks for telling me(It was a combination of eating food and histamine duplicating because of histamine production not stopping when dead)

MrMelbert reviewed (approved):
Approving this on the basis that, for the most part, these foods will inherent the reagents present from their ingredients (grown foods) rather than out of thin air -

If any food gimmicks are ruined by this, people can definitely feel free to re-add them in some way or another
